CRD docs generation failing

/kind bug


**What steps did you take and what happened:**

`make -C docs/book build` or `./docs/book/install-and-build.sh`

Both will fail this way:

```shell
% make -C docs/book build
./install-and-build.sh
downloading mdBook-v0.4.40-x86_64-apple-darwin.tar.gz
+ curl --retry 3 -sL -o /tmp/mdbook.tar.gz https://github.com/rust-lang/mdBook/releases/download/v0.4.40/mdBook-v0.4.40-x86_64-apple-darwin.tar.gz
+ tar -C /tmp -xzvf /tmp/mdbook.tar.gz
x mdbook
+ chmod +x /tmp/mdbook
+ genCRDAPIReferenceDocsPath=/tmp/gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb
+ genCRDAPIReferenceDocs=/tmp/gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/gen-crd-api-reference-docs
+ cd /tmp
+ curl --retry 3 -sL -o gen-crd-api-reference-docs.zip https://github.com/ahmetb/gen-crd-api-reference-docs/archive/642c3aa7441d324f54f5a9a6a1841ffffacf5aeb.zip
+ unzip gen-crd-api-reference-docs.zip
Archive:  gen-crd-api-reference-docs.zip
642c3aa7441d324f54f5a9a6a1841ffffacf5aeb
replace gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/.github/dependabot.yml? [y]es, [n]o, [A]ll, [N]one, [r]ename: A
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/.github/dependabot.yml  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/.github/workflows/ci.yml  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/.gitignore  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/.goreleaser.yml  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/LICENSE  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/README.md  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/example-config.json  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/go.mod  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/go.sum  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/main.go  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/template/members.tpl  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/template/pkg.tpl  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/template/placeholder.go  
  inflating: gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/template/type.tpl  
+ cd gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb
+ go build .
+ /tmp/gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/gen-crd-api-reference-docs -config /tmp/gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/example-config.json -template-dir /tmp/gen-crd-api-reference-docs-642c3aa7441d324f54f5a9a6a1841ffffacf5aeb/template -api-dir ./api/v1beta1 -out-file ./docs/book/src/reference/v1beta1-api-raw.html
I0414 09:45:12.678854   91363 main.go:144] parsing go packages in directory ./api/v1beta1
I0414 09:45:12.852083   91363 main.go:222] found 1 packages
F0414 09:45:14.800658   91363 main.go:150] no API packages found in ./api/v1beta1
make: *** [build] Error 255
```

Same thing happens in e2e with the Netlify deploy jobs.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

--- a/docs/book/install-and-build.sh
+++ b/docs/book/install-and-build.sh
@@ -24,7 +24,7 @@ cd "${KUBE_ROOT}" || exit 1
 os=$(go env GOOS)
 arch=$(go env GOARCH)
 mdBookVersion="v0.4.40"
-genCRDAPIReferenceDocsVersion="642c3aa7441d324f54f5a9a6a1841ffffacf5aeb"
+genCRDAPIReferenceDocsVersion="ee012b7016522a56d9cef3ea9baed3e0cfc8ff0e"
 
 # translate arch to rust's conventions (if we can)
 if [[ ${arch} == "amd64" ]]; then
@@ -64,19 +64,19 @@ ${cmd} /tmp/mdbook.${ext}
 chmod +x /tmp/mdbook
 
 # Generate API docs
-genCRDAPIReferenceDocsPath="/tmp/gen-crd-api-reference-docs-${genCRDAPIReferenceDocsVersion}"
-genCRDAPIReferenceDocs="${genCRDAPIReferenceDocsPath}/gen-crd-api-reference-docs"
+genCRDAPIReferenceDocsPath="/tmp/crddoc-${genCRDAPIReferenceDocsVersion}"
+genCRDAPIReferenceDocs="${genCRDAPIReferenceDocsPath}/crddoc document crds"
 (
   cd /tmp
-  curl --retry 3 -sL -o gen-crd-api-reference-docs.zip "https://github.com/ahmetb/gen-crd-api-reference-docs/archive/${genCRDAPIReferenceDocsVersion}.zip"
-  unzip gen-crd-api-reference-docs.zip
-  cd "gen-crd-api-reference-docs-${genCRDAPIReferenceDocsVersion}"
+  curl --retry 3 -sL -o crddoc.zip "https://github.com/theunrepentantgeek/crddoc/archive/${genCRDAPIReferenceDocsVersion}.zip"
+  unzip crddoc.zip
+  cd "crddoc-${genCRDAPIReferenceDocsVersion}"
   go build .
 )
 
-${genCRDAPIReferenceDocs} -config "${genCRDAPIReferenceDocsPath}/example-config.json" -template-dir "${genCRDAPIReferenceDocsPath}/template" -api-dir ./api/v1beta1 -out-file ./docs/book/src/reference/v1beta1-api-raw.html
-${genCRDAPIReferenceDocs} -config "${genCRDAPIReferenceDocsPath}/example-config.json" -template-dir "${genCRDAPIReferenceDocsPath}/template" -api-dir ./exp/api/v1beta1 -out-file ./docs/book/src/reference/v1beta1-exp-api-raw.html
-${genCRDAPIReferenceDocs} -config "${genCRDAPIReferenceDocsPath}/example-config.json" -template-dir "${genCRDAPIReferenceDocsPath}/template" -api-dir ./api/v1alpha1 -out-file ./docs/book/src/reference/v1alpha1-api-raw.html
+${genCRDAPIReferenceDocs} --config "${genCRDAPIReferenceDocsPath}/docs/config/crddoc-config.yaml" --template "${genCRDAPIReferenceDocsPath}/templates" --output ./docs/book/src/reference/v1beta1-api-raw.html ./api/v1beta1
+${genCRDAPIReferenceDocs} --config "${genCRDAPIReferenceDocsPath}/docs/config/crddoc-config.yaml" --template "${genCRDAPIReferenceDocsPath}/templates" --output ./docs/book/src/reference/v1beta1-exp-api-raw.html ./exp/api/v1beta1
+${genCRDAPIReferenceDocs} --config "${genCRDAPIReferenceDocsPath}/docs/config/crddoc-config.yaml" --template "${genCRDAPIReferenceDocsPath}/templates" --output ./docs/book/src/reference/v1alpha1-api-raw.html ./api/v1alpha1
 
 # Finally build the book.
 (cd docs/book && /tmp/mdbook build)

